src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -70,6 +70,9 @@ export class FileStat implements IFileStat {
 	public hasChildren: boolean;
 	public children: FileStat[];
 	public parent: FileStat;
+	public isVirtualDirectory: boolean;
+	public isVirtualDirectoryMember: boolean;
+	public virtualDirectoryName: string;
```

Comment:
In your implementation it's not used, but in #13754 it was used to override getId:

```
	public getId(tree: ITree, stat: FileStat): string {
  		if (this.enableVirtualDirectories && stat.isVirtualDirectoryMember) {
 			return stat.getId() + stat.virtualDirectoryName;
 		}
 
  		return stat.getId();
  	}
```

That's probably how they got around that treeModel situation from your last comment